commune.lic

```diff
@@ -0,0 +1,412 @@
+=begin
+  Documentation: ???
+=end
+
+custom_require.call(%w(common common-items common-arcana drinfomon))
+
+class Commune
+  include DRC
+  include DRCI
+  include DRCA
+
+  ######################## ITEM ACTIONS ########################
+
+  def get_cleric_item(item)
+    case bput("get #{item} from my #{@theurgy_supply_container}", 'You get', 'You are already holding', 'I could not find', 'What were you referring to')
+	when 'You get' || 'You are already holding'
+	  return true
+	else
+	  return false
+	end
+  end
+
+  def stow_cleric_item(item)
+    case bput("put my #{item} in my #{@theurgy_supply_container}", 'You put', 'What were you referring to')
+	when 'You put'
+	  return true
+	else
+	  return false
+	end
+  end
+
+  def empty_cleric_hands(hand='both')
+    if hand == 'both'
+	  empty_cleric_right_hand
+	  empty_cleric_left_hand
+	elsif hand == 'right'
+	  empty_cleric_right_hand
+	elsif hand == 'left'
+	  empty_cleric_left_hand
+	end
+  end
+  
+  def empty_cleric_right_hand()
+  	 stow_cleric_item(GameObj.right_hand.noun) if @cleric_items.include?(GameObj.right_hand.name)
+	 stow_cleric_item(GameObj.right_hand.noun) if @cleric_items.include?(GameObj.right_hand.noun)
+	 fput('stow right') if GameObj.right_hand.noun != nil
+  end
+  
+  def empty_cleric_left_hand()
+  	 stow_cleric_item(GameObj.left_hand.noun) if @cleric_items.include?(GameObj.left_hand.name)
+	 stow_cleric_item(GameObj.left_hand.noun) if @cleric_items.include?(GameObj.left_hand.noun)
+	 fput('stow left') if GameObj.left_hand.noun != nil
+  end
+  
+  ######################## ITEM CHECKS ########################
+
+  def have_holy_water?()
+	case bput("count holy water in my #{@water_holder}", 'left of the holy water', 'I could not find', 'What were you referring to')
+	when 'left of the holy water'
+	  return true
+	else
+	  echo('No holy water found. Please make sure containers are open.')
+	  missing_item('holy water')
+	end
+  end
+
+  def have_cleric_item?(item)
+    case bput("look at my #{item} in my #{@theurgy_supply_container}", 'You see nothing unusual', 'I could not find', 'What were you referring to')
+    when 'You see nothing unusual'
+	  return true
+    else
+	  echo("No #{item} found. Please make sure containers are open.")
+	  missing_item("#{item}")
+    end
+  end
+  
+  def holding?(noun)
+    if GameObj.right_hand.noun == "#{noun}"
+	  return true
+	elsif GameObj.left_hand.noun == "#{noun}"
+	  return true
+	else
+	  return false
+	end
+  end
+  
+  # must pass items in an array
+  def have_cleric_items?(items)
+    items.each do |item|
+	  if
+	    item == 'holy water' then have_holy_water?
```

Comment:
I noticed that `have_holy_water?` and `have_cleric_item?` are predicates, yet their return value isn't used. Is the intent here that we exit if an item is missing?

Reply:
the have_holy_water? method that is called will check if holy water exists and, if it doesn't, it will exit the script. if have_holy_water? returns true (holy water DOES exist) then the script continues to run through each |item| and make sure we have the necessary items for the commune.